src/main/java/org/ibp/api/java/impl/middleware/study/StudyEntryServiceImpl.java

```diff
@@ -91,13 +117,60 @@ public StudyEntryDto replaceStudyEntry(final Integer studyId, final Integer entr
 			.replaceStudyEntry(studyId, entryId, gid, this.pedigreeService.getCrossExpansion(gid, this.crossExpansionProperties));
 	}
 
+	@Override
+	public List<StudyEntryDto> createStudyEntries(final Integer studyId,
+		final StudyEntryGeneratorBatchRequestDto studyEntryGeneratorBatchRequestDto, final String programUuid) {
+		this.studyValidator.validate(studyId, true);
+		if(studyEntryGeneratorBatchRequestDto.getListId() != null && studyEntryGeneratorBatchRequestDto.getListId() != 0) {
+			return this.createStudyEntries(studyId, studyEntryGeneratorBatchRequestDto.getListId());
+		}
+
+		//Validate EntryType
+		this.entryTypeValidator.validateEntryType(studyEntryGeneratorBatchRequestDto.getEntryTypeId(), programUuid);
+
+		final SearchCompositeDto<Integer, Integer> searchComposite = studyEntryGeneratorBatchRequestDto.getSearchComposite();
+		final BindingResult errors = new MapBindingResult(new HashMap<>(), LotGeneratorBatchRequestDto.class.getName());
+		this.searchCompositeDtoValidator.validateSearchCompositeDto(searchComposite, errors);
+		final List<Integer> gids = this.searchRequestDtoResolver.resolveGidSearchDto(searchComposite);
+		this.germplasmValidator.validateGids(errors, gids);
+
+		final List<Germplasm> germplasmList = this.germplasmDataManager.getGermplasmWithPreferredNameByGIDs(gids);
+		final ModelMapper mapper = StudyEntryMapper.getInstance();
+		//Create StudyEntryDtos from given gids
+		final List<StudyEntryDto> studyEntryDtoList =
+			germplasmList.stream().map(l -> mapper.map(l, StudyEntryDto.class)).collect(Collectors.toList());
+
+
+		final List<Integer> germplasmDescriptorIds = this.getEntryDescriptorColumns(studyId).stream()
+			.map(measurementVariable -> measurementVariable.getTermId()).collect(Collectors.toList());
+		final Map<Integer, Germplasm> germplasmMap =
+			germplasmList.stream().collect(Collectors.toMap(Germplasm::getGid, g -> g));
+
+		//Get the next entry number
+		Integer entryNumber = this.middlewareStudyEntryService.getNextEntryNumber(studyId);
+		final boolean hasCrossGermplasmDescriptor = germplasmDescriptorIds.contains(TermId.CROSS.getId());
+		for(final StudyEntryDto studyEntryDto: studyEntryDtoList) {
+			//Retrieve cross value only if the CROSS is present in the germplasm descriptors
+			final String cross = hasCrossGermplasmDescriptor ?
+				this.pedigreeService.getCrossExpansion(studyEntryDto.getGid(), this.crossExpansionProperties) : "";
```

Comment:
Retrieve cross expansion strings for all GIDs at once (if cross descriptor is present), instead of inside the loop